File: Exemplos M10/App_IntentDigitalHUB_XamarinAndroid/Elgin_IntentDigitalHUB_XamarinAndroid-font/Balanca/BalanceProtocol.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using AndroidX.Annotations;

/**
 * Protocolos de comunicação com a balança, o código de cada protocolo corresponde ao índice de declaração no enum
 * Ex: Protocolo 4 possui 4 de código correspondente
 */

namespace Xamarin_Android_Intent_Digital_Hub.Balanca
{
    class BalanceProtocol
    {
        private BalanceProtocol(string friendlyName) { this.friendlyName = friendlyName; }

        private String friendlyName;

        public string toString() { return this.friendlyName; }

        public static BalanceProtocol PROTOCOL_0 = new BalanceProtocol("PROTOCOL 0");
        public static BalanceProtocol PROTOCOL_1 = new BalanceProtocol("PROTOCOL 1");
        public static BalanceProtocol PROTOCOL_2 = new BalanceProtocol("PROTOCOL 2");
        public static BalanceProtocol PROTOCOL_3 = new BalanceProtocol("PROTOCOL 3");
        public static BalanceProtocol PROTOCOL_4 = new BalanceProtocol("PROTOCOL 4");
        public static BalanceProtocol PROTOCOL_5 = new BalanceProtocol("PROTOCOL 5");
        public static BalanceProtocol PROTOCOL_6 = new BalanceProtocol("PROTOCOL 6");
        public static BalanceProtocol PROTOCOL_7 = new BalanceProtocol("PROTOCOL 7");
    }

}

File: Exemplos M10/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_Flutter-font/lib/pages/balanca/balance_protocol.dart
// ignore_for_file: constant_identifier_names

part of 'balance_page.dart';

enum BalanceProtocol {
  PROTOCOL_0,
  PROTOCOL_1,
  PROTOCOL_2,
  PROTOCOL_3,
  PROTOCOL_4,
  PROTOCOL_5,
  PROTOCOL_6,
  PROTOCOL_7,
}

extension BalanceProtocolExtension on BalanceProtocol {
  //Define o @name de cada protocol afim de reusar na apresentação da tela
  String get name {
    switch (this) {
      case BalanceProtocol.PROTOCOL_0:
        return 'PROTOCOL 0';
      case BalanceProtocol.PROTOCOL_1:
        return 'PROTOCOL 1';
      case BalanceProtocol.PROTOCOL_2:
        return 'PROTOCOL 2';
      case BalanceProtocol.PROTOCOL_3:
        return 'PROTOCOL 3';
      case BalanceProtocol.PROTOCOL_4:
        return 'PROTOCOL 4';
      case BalanceProtocol.PROTOCOL_5:
        return 'PROTOCOL 5';
      case BalanceProtocol.PROTOCOL_6:
        return 'PROTOCOL 6';
      case BalanceProtocol.PROTOCOL_7:
        return 'PROTOCOL 7';
    }
  }

  //Retorna o código de protocolo correspondente de cada protocolo
  int get protocolCode {
    switch (this) {
      case BalanceProtocol.PROTOCOL_0:
        return 0;
      case BalanceProtocol.PROTOCOL_1:
        return 1;
      case BalanceProtocol.PROTOCOL_2:
        return 2;
      case BalanceProtocol.PROTOCOL_3:
        return 3;
      case BalanceProtocol.PROTOCOL_4:
        return 4;
      case BalanceProtocol.PROTOCOL_5:
        return 5;
      case BalanceProtocol.PROTOCOL_6:
        return 6;
      case BalanceProtocol.PROTOCOL_7:
        return 7;
    }
  }
}
